File: backend/src/main/java/com/clinassist/dto/CreateSeanceRequest.java
package com.clinassist.dto;

import com.clinassist.entity.Seance;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class CreateSeanceRequest {
    
    @NotNull(message = "Patient ID is required")
    private Long patientId;
    
    @NotNull(message = "Therapeute ID is required")
    private Long therapeuteId;
    
    @NotNull(message = "Scheduled date is required")
    @Future(message = "Scheduled date must be in the future")
    private LocalDateTime scheduledAt;
    
    private Integer durationMinutes = 60;
    
    private Seance.SeanceType type = Seance.SeanceType.IN_PERSON;
    
    private String objectives;
    private String notes;
    
    private String meetingRoom;
    
    private Boolean isRecurring = false;
    private String recurringPattern;
}

File: mobile/lib/features/appointments/data/models/create_seance_request.dart
class CreateSeanceRequest {
  final int patientId;
  final int therapeuteId;
  final DateTime scheduledAt;
  final int durationMinutes;
  final String type;
  final String? initialStatus;
  final String? objectives;
  final String? notes;

  CreateSeanceRequest({
    required this.patientId,
    required this.therapeuteId,
    required this.scheduledAt,
    this.durationMinutes = 60,
    this.type = 'IN_PERSON',
    this.initialStatus,
    this.objectives,
    this.notes,
  });

  Map<String, dynamic> toJson() {
    return {
      'patientId': patientId,
      'therapeuteId': therapeuteId,
      'scheduledAt': scheduledAt.toIso8601String(),
      'durationMinutes': durationMinutes,
      'type': type,
      if (initialStatus != null) 'initialStatus': initialStatus,
      if (objectives != null) 'objectives': objectives,
      if (notes != null) 'notes': notes,
    };
  }
}
